src/test/kotlin/no/nav/syfo/oppfolgingstilfelle/kafka/KafkaOppfolgingstilfelleArbeidstakerServiceSpek.kt

```diff
@@ -36,31 +37,65 @@ class KafkaOppfolgingstilfelleArbeidstakerServiceSpek : Spek({
+            kafkaOppfolgingstilfelleArbeidstakerDialogmotekandidatFirst,
+        )
+        val kafkaOppfolgingstilfelleArbeidstakerDialogmotekandidatFirstRecordDuplicate = ConsumerRecord(
+            OPPFOLGINGSTILFELLE_ARBEIDSTAKER_TOPIC,
+            partition,
+            1,
+            "key1",
+            kafkaOppfolgingstilfelleArbeidstakerDialogmotekandidatFirst,
+        )
+        val kafkaOppfolgingstilfelleArbeidstakerNotDialogmotekandidat = generateKafkaOppfolgingstilfelleArbeidstaker(
+            arbeidstakerPersonIdentNumber = ARBEIDSTAKER_PERSONIDENTNUMBER,
+            durationDays = DIALOGMOTEKANDIDAT_STOPPUNKT_DURATION_DAYS - 1
+        )
+        val kafkaOppfolgingstilfelleArbeidstakerNotDialogmotekandidatRecord = ConsumerRecord(
+            OPPFOLGINGSTILFELLE_ARBEIDSTAKER_TOPIC,
+            partition,
+            2,
+            "key2",
+            kafkaOppfolgingstilfelleArbeidstakerNotDialogmotekandidat,
+        )
+        val kafkaOppfolgingstilfelleArbeidstakerDialogmotekandidatLast = generateKafkaOppfolgingstilfelleArbeidstaker(
+            arbeidstakerPersonIdentNumber = ARBEIDSTAKER_PERSONIDENTNUMBER,
+            durationDays = DIALOGMOTEKANDIDAT_STOPPUNKT_DURATION_DAYS + 1
+        )
+        val kafkaOppfolgingstilfelleArbeidstakerDialogmotekandidatLastRecord = ConsumerRecord(
+            OPPFOLGINGSTILFELLE_ARBEIDSTAKER_TOPIC,
+            partition,
+            3,
+            "key3",
+            kafkaOppfolgingstilfelleArbeidstakerDialogmotekandidatLast,
         )
 
         val mockKafkaConsumerOppfolgingstilfelleArbeidstaker =
             mockk<KafkaConsumer<String, KafkaOppfolgingstilfelleArbeidstaker>>()
         every { mockKafkaConsumerOppfolgingstilfelleArbeidstaker.poll(any<Duration>()) } returns ConsumerRecords(
             mapOf(
                 oppfolgingstilfelleArbeidstakerTopicPartition to listOf(
-                    kafkaOppfolgingstilfelleArbeidstakerRecord,
+                    kafkaOppfolgingstilfelleArbeidstakerDialogmotekandidatFirstRecord,
+                    kafkaOppfolgingstilfelleArbeidstakerDialogmotekandidatFirstRecordDuplicate,
+                    kafkaOppfolgingstilfelleArbeidstakerNotDialogmotekandidatRecord,
+                    kafkaOppfolgingstilfelleArbeidstakerDialogmotekandidatLastRecord,
                 )
             )
         )
         every { mockKafkaConsumerOppfolgingstilfelleArbeidstaker.commitSync() } returns Unit
 
         describe("${KafkaOppfolgingstilfelleArbeidstakerService::class.java.simpleName}: Poll and proceess reccords") {
             describe("Receive records, store and retrieve oppfolgingstilfelleArbeidstaker for PersonIdent") {
-                it("should return OppfolgingstilfelleArbeidstaker created from KafkaOppfolgingstilfelleArbeidstaker") {
+
+                it("should create OppfolgingstilfelleArbeidstaker exactly once if Dialogmotekandidat and not already created(skip duplicates)") {
```

Comment:
Testen sier exactly once, mens det sjekkes på size = 2 lenger ned. Kanskje heller lage flere tester én for å sjekke duplikat-håntering og én som sjekker at man lagrer to hvis det kommer to ulike?

Reply:
Fint det. Nå har jeg delt opp i 2 tester: 1 for duplikat-testing og en for lagring/sjekking av flere unike.